In [8]:
import numpy as np

In [9]:
import scipy.stats as stats

**Задача №1.** Вы провели эксперимент c упрощением формы заказа в магазине Утконос и получили результаты по метрике конверсий в покупку. Выберите метод оценки и оцените есть ли стат.значимые различия между конверсиями в двух группах при alpha = 5%. Дайте краткие рекомендации команде. <br>
Результаты:
1. Число юзеров в группах , которые заходили на сайт в период эксперимента: n1 = 15550 и n2 = 15550.
2. Число юзеров в группах , которые совершили хотя бы одну покупку за период эксперимента: m1 = 164 и m2 = 228.
3. Конверсии : conv1 = 1.05% conv2 = 1.47%.

**Решение:** <br>
Для оценки статистической значимости различий между конверсиями в двух группах будем использовать Z-критерий для разности долей:\
$Z = \cfrac{p_1 - p_2}{\sqrt{p (1-p) \left(\cfrac{1}{n_1} + \cfrac{1}{n_2}\right)}}$, где
$p = \cfrac{m_1 + m_2}{n_1 + n_2}$,


In [10]:
n1=15550
n2=15550
m1=164
m2=228
alpha = 0.05

In [14]:
p1=m1/n1
print(round(p1,4))

0.0105


In [15]:
p2=m2/n2
print(round(p2,4))

0.0147


In [17]:
p=(m1+m2)/(n1+n2)
print(round(p,4))

0.0126


In [20]:
Z = (p1-p2)/np.sqrt(p*(1-p)*(1/n1+1/n2))
print(round(Z,4))


-3.2531


**Ответ:** <br>
Для уровня значимости alpha = 5% критическое значение Z-критерия равно 1.96. Так как полученное расчетное значение Z-критерия меньше критического значения -3.2531 < 1.96, то мы отвергаем нулевую гипотезу о равенстве конверсий в группах. И делаем вывод, что эксперимент с упрощением формы заказа в магазине
Утконос дал статистически значимый эффект на уровень конверсии в покупку. Рекомендация команде: использовать новую форму заказа.

**Задача №2.** Сравниваем метрику конверсия в покупку. Размер выборки - 10000 элементов в каждой группе. Какой статистический критерий тут лучше всего подойдёт и почему?

**Ответ:** <br>
Лучше всего для сравнения метрики конверсия в покупку подойдет Z-критерий разности долей. Так как в отличие от t-критерия (размер выборки до 30 элементов), Z-критерий долей может использоваться как для больших (более 30 элементов), так и для малых выборок, поскольку он основан на распределении Бернулли и не требует нормального распределения данных.